safe/converter.py

```diff
@@ -327,7 +323,8 @@ def encoder(
             )
 
         scaffold_str = ".".join(frags_str)
-        attach_pos = set(re.findall(r"(\[\d+\*\]|\[[^:]*:\d+\])", scaffold_str))
+        # don't capture atom mapping in the scaffold
+        attach_pos = set(re.findall(r"(\[\d+\*\]|!\[[^:]*:\d+\])", scaffold_str))
```

Comment:
The problem here is that if we don't handle the atom mapping, the `attach_pos` set will contain the atom mapping pieces and will give us wrong molecules in some cases.

As an example:
```python
smiles="c1cc2c(cc1[C@@H]1CC[C:3][NH2+]1)O[13C]CO2"
scaffold_str = ".".join(frags_str) # c1cc2c(cc1[1*])O[13C]CO2.[C@@H]1([1*])CC[C:3][NH2+]1
attach_pos = set(re.findall(r"(\[\d+\*\]|\[[^:]*:\d+\])", scaffold_str)) #  #{'[13C]CO2.[C@@H]1([1*])CC[C:3]', '[1*]'}
``` 
this will give us the safe string `'c1cc2c(cc13)O[13C]CO2.[C@@H]13CC[C:3][NH2+]1'`

With my regex:
```python
smiles="c1cc2c(cc1[C@@H]1CC[C:3][NH2+]1)O[13C]CO2"
scaffold_str = ".".join(frags_str) # c1cc2c(cc1[1*])O[13C]CO2.[C@@H]1([1*])CC[C:3][NH2+]1
attach_pos = set(re.findall(r"(\[\d+\*\]|\[[^:]*:\d+\])", scaffold_str)) #  {'[1*]'}
``` 
that will give us in this case the exact safe string `'c1cc2c(cc13)O[13C]CO2.[C@@H]13CC[C:3][NH2+]1'`

For more difficult cases of atom mapping like `[C+:1]#[C:2]CC(C[C:20][O:21]CN)C[C:3]1=[C:7]([H:10])[N-:6][O:5][C:4]1([H:8])[H:9]"`
![image](https://github.com/datamol-io/safe/assets/30401800/5f9a443f-b6da-404a-a480-2daaf98f82fd)

We will have the following attach_pos and final safe string:
```python
modified_regex = {'[1*]'}
old_regex = {'[C:3]', '[N-:6]', '[C:4]', '[O:21]', '[H:9]', '[H:8]', '[C:7]', '[C+:1]', '[C:2]', '[1*]', '[O:5]', '[H:10]', '[C:20]'}
# modified regex
'[C+:1]#[C:2]CC(C[C:3]1=[C:7]([H:10])[N-:6][O:5][C:4]1([H:8])[H:9])C[C:20]2.[O:21]2CN'
# old regex
"[C+:1]#[C:2]CC(C[C:3]1=[C:7]([H:10])[N-:6][O:5][C:4]1([H:8])[H:9])C[C:20][1*].[O:21]([1*])CN"
``` 
That will lead at the following mols:
`old regex`
![image](https://github.com/datamol-io/safe/assets/30401800/3219ead8-1915-43d2-b3d4-b376420f5e45)
`new regex`
![image](https://github.com/datamol-io/safe/assets/30401800/bde29006-0835-40b0-ba1d-eb26969cbfc2)